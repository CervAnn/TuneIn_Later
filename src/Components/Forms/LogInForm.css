.LogInFormContainer {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    font-family: 'Permanent Marker', cursive;
    color: #F4E697;
}

form {
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    width: 35%;
    margin: auto;
    height: 170px;
}

input {
    height: 50px;
    font-size: 20px;
    border-radius: 15px;
    padding: 5px;
}