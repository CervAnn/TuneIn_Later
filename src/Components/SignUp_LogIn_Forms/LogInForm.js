import React, { Component } from 'react'

class LogInForm extends Component {
    constructor() {
        super();
        this.state = {
            email: "",
            password: ""
        }
    }

    handleChange = () => {
        this.setState({[e.target.name]: e.target.value})
    }

    handleSubmit = () => {
        e.preventDefault()
        let currentUser = {

        }
    }

    clearInputs = () => {

    }

    render() {
        return (
            <form className="LogInForm">
                <input
                type="text"
                placeholder="Email: "
                name="email"
                value={this.state.email}
                onChange={this.handleChange}
                />
                <input
                type="text"
                placeholder="Password: "
                name="password"
                value={this.state.password}
                onChange={this.handleChange}
                />
                <button onClick={}>Submit</button>
            </form>
        )
    }
}

export default LogInForm;


