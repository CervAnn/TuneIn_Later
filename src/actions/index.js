export const addAlbums = (albums) => ({
	type: 'ADD_ALBUMS',
	albums
});

export const validUser = (user) => ({
	type: 'VALID_USER',
	user
});

export const addFavorite = (id) => ({
	type: 'ADD_FAVORITE',
	id
});

export const invalidUser = () => ({
	type: 'INVALID_USER'
});

export const getUserFavorites = (userFavorites) => ({
	type: 'GET_USER_FAVORITES',
	userFavorites
});