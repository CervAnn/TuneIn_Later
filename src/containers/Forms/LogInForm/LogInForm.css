.LogInFormContainer {
    align-items: center;
    color: #F4E697;
    display: flex;
    flex-direction: column;
    font-family: 'Permanent Marker', cursive;
    justify-content: center;
}

form {
    display: flex;
    flex-direction: column;
    height: 170px;
    justify-content: space-evenly;
    margin: auto;
    width: 35%;
}

input {
    border-radius: 15px;
    font-size: 20px;
    height: 50px;
    padding: 5px;
}

.loginUser {
    border-radius: 15px;
    color: #F4E697;
    font-size: 40px;
}

.loginUser:hover, #signup-link:hover {
    border-radius: 15px;
    border: 2px solid #cc7b3a;
}



