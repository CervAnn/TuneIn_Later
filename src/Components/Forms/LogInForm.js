import React, { Component } from 'react';
import { fetchUser } from '../../util/apiCalls';
import { validUser } from '../../actions/index';
import { connect } from 'react-redux';
import { Redirect, Link } from 'react-router-dom';
import './LogInForm.css'


class LogInForm extends Component {
	constructor() {
		super();
		this.state = {
			email: "",
			password: "",
			error: ""
		}
	}

	handleChange = (e) => {
		this.setState({[e.target.name]: e.target.value})
	}

	handleSubmit = (e) => {
		e.preventDefault()
		let user = {
				email: this.state.email,
				password: this.state.password
		}
		this.userLogin(user);
		this.clearInputs();
	}

	userLogin = (user) => {
		fetchUser(user)
		.then(user => this.props.validUser(user))
		.catch(error => this.setState({error}))
	}

	clearInputs = () => {
		this.setState({email: '', password: ''});
	}

	render() {
		if (this.props.user) {
			return <Redirect to="/" />
		}

		return (
			<article className="LogInFormContainer">
				<form className="LogInForm">
					<input
					type="text"
					placeholder="Email: "
					name="email"
					value={this.state.email}
					onChange={this.handleChange}
					required/>
					<input
					type="text"
					placeholder="Password: "
					name="password"
					value={this.state.password}
					onChange={this.handleChange}
					required/>
					<button 
						onClick={event => this.handleSubmit(event)}
					>
						Submit
					</button>
				</form>
				<p>Are you a new user? <Link to="/signup">Sign Up Here</Link></p>
				{this.state.error && <p>The email and/or password do not match an existing user.</p>}
			</article>
		)
	}
}

const mapStateToProps = (state) => ({
	user: state.user
})

const mapDispatchToProps = (dispatch) => ({
	validUser: (user) => dispatch(validUser(user))
});

export default connect(mapStateToProps, mapDispatchToProps)(LogInForm);


