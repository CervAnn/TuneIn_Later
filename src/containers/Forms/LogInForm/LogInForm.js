import React, { Component } from 'react';
import { fetchUser, fetchUserFavorites } from '../../../util/apiCalls';
import { validUser, getUserFavorites } from '../../../actions/index';
import { connect } from 'react-redux';
import { Redirect, Link } from 'react-router-dom';
import './LogInForm.css'


class LogInForm extends Component {
	constructor() {
		super();
		this.state = {
			email: "",
			password: "",
			error: ""
		}
	}

	handleChange = (e) => {
		this.setState({[e.target.name]: e.target.value})
	}

	handleSubmit = (e) => {
		e.preventDefault()
		let user = {
				email: this.state.email,
				password: this.state.password
		}
		this.userLogin(user);
		this.clearInputs();
	}

	userLogin = (user) => {
		fetchUser(user)
		.then(user => this.props.validUser(user))
		.then(data => data.user.id ? fetchUserFavorites(data.user.id) : null)
    .then(data => this.props.getUserFavorites(data.favorites))
		.catch(error => this.setState({error}))
	}

	clearInputs = () => {
		this.setState({email: '', password: ''});
	}

	render() {
		if (this.props.user) {
			return <Redirect to="/" />
		}

		return (
			<article className="LogInFormContainer">
				<form className="LogInForm">
					<input
					type="text"
					placeholder="Email: "
					name="email"
					value={this.state.email}
					onChange={this.handleChange}
					required/>
					<input
					type="text"
					placeholder="Password: "
					name="password"
					value={this.state.password}
					onChange={this.handleChange}
					required/>
					<button className="loginUser"
						onClick={event => this.handleSubmit(event)}
					>
						Submit
					</button>
				</form>
				<p>Are you a new user? <Link to="/signup">Sign Up Here</Link></p>
				{this.state.error && <p id="error">The email and/or password do not match an existing user.</p>}
			</article>
		)
	}
}

const mapStateToProps = (state) => ({
	user: state.user
})

const mapDispatchToProps = (dispatch) => ({
    validUser: (user) => dispatch(validUser(user)),
    getUserFavorites: (favorites) => dispatch(getUserFavorites(favorites))
});

export default connect(mapStateToProps, mapDispatchToProps)(LogInForm);


