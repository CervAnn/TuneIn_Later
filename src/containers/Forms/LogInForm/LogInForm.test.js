import React from 'react';
import { shallow } from 'enzyme';
import { LogInForm, mapStateToProps, mapDispatchToProps } from './LogInForm';
import { validUser, getUserFavorites } from '../../../actions/index'

describe('LogInForm', () => {
  let wrapper, mockUser, mockValidUser, mockGetUserFavorites;

  beforeEach(() => {
    mockUser = null;
    mockValidUser = jest.fn();
    mockGetUserFavorites = jest.fn();
    wrapper = shallow(<LogInForm 
    user={mockUser}
    validUser={mockValidUser}
    getUserFavorites={mockGetUserFavorites}
    />);
  });

  it('should match the snapshot with all data passed in correctly', () => {

    expect(wrapper).toMatchSnapshot();
  });

  it('should update state on a change in an input', () => {

    const mockEvent = {target: {name: 'email', value: 'alan@turing.io'}}

    wrapper.find('input').at(0).simulate('change', mockEvent);

    expect(wrapper.state('email')).toEqual('alan@turing.io')
  });

  it('should update state on a change in an input', () => {

    const mockEvent = { target: { name: 'password', value: 'password' } }
   
    wrapper.find('input').at(1).simulate('change', mockEvent);

    expect(wrapper.state('password')).toEqual('password');
  });

  it('should call handleSubmit on a button click', () => {

    const mockEvent = {preventDefault: jest.fn()}

    wrapper.instance().handleSubmit = jest.fn();
    wrapper.instance().forceUpdate();
    wrapper.find('button').simulate('click', mockEvent);

    expect(wrapper.instance().handleSubmit).toHaveBeenCalled();
  });

  it('should call user log in on a button click', () => {
    const mockEvent = { preventDefault: jest.fn() }

    wrapper.instance().userLogin = jest.fn();
    wrapper.instance().forceUpdate();
    wrapper.find('button').simulate('click', mockEvent);

    expect(wrapper.instance().userLogin).toHaveBeenCalled();
  });

  it('should update the state when clearInputs is called', () => {

    const expected = '';

    wrapper.setState({
      email: 'alan@turing.io',
      password: 'password',
      error: 'Wrong'
    });

    wrapper.instance().clearInputs();

    expect(wrapper.state('email')).toEqual(expected);
    expect(wrapper.state('password')).toEqual(expected);
    expect(wrapper.state('error')).toEqual(expected);
  });

});

describe('mapStateToProps', () => {

  it('should return an object with the current user', () => {
    const mockState = {
      albums: [],
      user: { id: 1, name: 'Alan', email: 'alan@turing.io' },
      error: '',
      favorites: [
        {
          album_id: 626204707,
          artist_name: 'Beyoncé',
          album_name: '4 (Expanded Edition)',
          price: 11.99,
          artwork_url: 'https://is2-ssl.mzstatic.com/image/thumb/Music6/v4/17/84/3a/17843a6d-1f2b-7e1e-a39f-3ff865110993/source/100x100bb.jpg',
          release_date: '2013-12-13T08:00:00Z',
          content_advisory_rating: 'Explicit',
          primary_genre_name: 'Pop',
        }
      ]
    }

    const expected = {
      user: { id: 1, name: 'Alan', email: 'alan@turing.io' }
    }

    const mappedProps = mapStateToProps(mockState);

    expect(mappedProps).toEqual(expected)
  });
});

describe('mapDispatchToProps', () => {

  it('should dispatch with a validUser with validUser is called', () => {
    const mockUser = { id: 1, name: 'Alan', email: 'alan@turing.io' }
    const mockDispatch = jest.fn();
    const actionToDispatch = validUser(mockUser);
  
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.validUser({ id: 1, name: 'Alan', email: 'alan@turing.io' });
  
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  });

  it('should dispatch with a validUser with validUser is called', () => {
    const mockFavorites = [
        {
          album_id: 626204707,
          artist_name: 'Beyoncé',
          album_name: '4 (Expanded Edition)',
          price: 11.99,
          artwork_url: 'https://is2-ssl.mzstatic.com/image/thumb/Music6/v4/17/84/3a/17843a6d-1f2b-7e1e-a39f-3ff865110993/source/100x100bb.jpg',
          release_date: '2013-12-13T08:00:00Z',
          content_advisory_rating: 'Explicit',
          primary_genre_name: 'Pop',
        }
      ]
  });
});