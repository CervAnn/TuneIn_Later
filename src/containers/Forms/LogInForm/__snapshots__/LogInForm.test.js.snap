// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogInForm should match the snapshot with all data passed in correctly 1`] = `
<article
  className="LogInFormContainer"
>
  <form
    className="LogInForm"
  >
    <input
      name="email"
      onChange={[Function]}
      placeholder="Email: "
      required={true}
      type="text"
      value=""
    />
    <input
      name="password"
      onChange={[Function]}
      placeholder="Password: "
      required={true}
      type="text"
      value=""
    />
    <button
      className="loginUser"
      onClick={[Function]}
    >
      Submit
    </button>
  </form>
  <p
    id="signup-link"
  >
    Are you a new user? 
    <Link
      to="/signup"
    >
      Click here to sign up!
    </Link>
  </p>
</article>
`;
